spark/v3.3/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestRewriteManifestsProcedure.java

```diff
@@ -75,6 +96,79 @@ public void testRewriteLargeManifests() {
         "Must have 4 manifests", 4, table.currentSnapshot().allManifests(table.io()).size());
   }
 
+  @Test
+  public void testRewriteLargeManifestsWithDateOrTimestampPartitionedTables() {
+    String[] scenarios = new String[] {"true", "false"};
+    String[] partitionColTypes = new String[] {"DATE", "TIMESTAMP"};
+    for (String scenario : scenarios) {
```

Comment:
Thanks for adding now. Testcase cover all the scenarios. 

Instead of these two for loops, we can use Junit's `ParameterizedTest` with `CsvSource`, it will be very clean compared to this. (as we already depend on Junit5)

Example:
https://github.com/projectnessie/nessie/blob/main/clients/spark-extensions-basetests/src/main/java/org/projectnessie/spark/extensions/AbstractNessieSparkSqlExtensionTest.java#L589-L596

Reply:
> Instead of these two for loops, we can use Junit's ParameterizedTest with CsvSource, it will be very clean compared to this. (as we already depend on Junit5)

+1 on the suggestion

I tried making this UT a parameterized test as per last feedback, but this UT didn't ran when i ran the complete suite, I think this is happening due to `SparkCatalogTestBase` already running with a parameterized class and we extending that.
https://github.com/apache/iceberg/blob/a8b67085e671bb5fa5a7f738ce2e9cb2dcc65d9c/spark/v3.3/spark/src/test/java/org/apache/iceberg/spark/SparkCatalogTestBase.java#L27

Let me dig in more.

Reply:
I'd be fine using `@ParamterizedTest`, but it would also work to just have two test cases that both enable Java 8 date/time types.